pkg/engine/helpers/object_key.go

```diff
@@ -0,0 +1,62 @@
+package helpers
```

Comment:
I had to extract these methods into a separate package because of circular dependencies 😞